### ISSUE TITLE
Don't use Rails.application.assets

### ISSUE THREAD

**Issue by sepastian** [spree-contrib/spree_print_invoice#105]
I have updated sprockets and now get `ActionView::Template::Error (undefined method `find_asset' for nil:NilClass)`.

`Rails.application.assets` is `nil` unless `assets.compile` is set, which defaults to `false` in production.

See https://github.com/rails/sprockets-rails/issues/237, i.e.:

> So it sounds like using those gems may have been behaving poorly in production before.

**Comment by mdavo6** [comment]
I had the same issue. Was able to resolve by changing the first few lines of _header.pdf.prawn to:
```
rel_path = Rails.application.assets_manifest.assets[Spree::PrintInvoice::Config[:logo_path]]
im = Rails.root.join("public","assets",rel_path)

if im && File.exist?(im)
  pdf.image im, vposition: :top, height: 40, scale: Spree::PrintInvoice::Config[:logo_scale]
end
```

**Comment by aman29april** [comment]
@mdavo6  This solution doesn't work in the development environment.

**Comment by mdavo6** [comment]
@aman29april Agree. Have just had some time to work on a better solution which works in production and development. What about the below as an approach? Working for me in development and production.

    if Rails.env.production? && 
    Rails.application.assets_manifest.assets[Spree::PrintInvoice::Config[:logo_path]].present?
      manifest_dir = Rails.application.assets_manifest.dir
      logo_path = Rails.application.assets_manifest.assets[Spree::PrintInvoice::Config[:logo_path]]
      im = File.join(manifest_dir, logo_path)
    elsif Rails.application.assets.find_asset(Spree::PrintInvoice::Config[:logo_path]) != nil
      im = Rails.application.assets.find_asset(Spree::PrintInvoice::Config[:logo_path]).filename
    end

    if im && File.exist?(im)
      pdf.image im, vposition: :top, height: 40, scale: Spree::PrintInvoice::Config[:logo_scale]
    end
